ConfigSpace/configuration_space.pyx

```diff
@@ -744,6 +747,9 @@ class ConfigurationSpace(object):
         """
         return list(self._hyperparameters.keys())
 
+    def __getitem__(self, key: str) -> Hyperparameter:
+        return self._hyperparameters[key]
```

Comment:
What do you think about calling `get_hyperparameter` here?

Reply:
Makes sense, has the validation checks